keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -1125,6 +1147,18 @@ class SqlDeliveryConfigRepository(
         }
     }
 
+  override fun deliveryConfigLastChecked(deliveryConfig: DeliveryConfig): Instant {
+    return sqlRetry.withRetry(READ) {
+      jooq
+        .select(DELIVERY_CONFIG_LAST_CHECKED.AT)
+        .from(DELIVERY_CONFIG_LAST_CHECKED, DELIVERY_CONFIG)
+        .where(DELIVERY_CONFIG_LAST_CHECKED.DELIVERY_CONFIG_UID.eq(DELIVERY_CONFIG.UID))
+        .and(DELIVERY_CONFIG.NAME.eq(deliveryConfig.name))
+        .fetchOne(DELIVERY_CONFIG_LAST_CHECKED.AT)
+        .toInstant(UTC)
+    }
+  }
```

Comment:
New method to just get the timestamp when the delivery config was last check for environment promotions

Reply:
It looks like a precondition of this function is that there exists a record in the delivery_config_last_checked table for that delivery config. Will that always be true? 

If not, I think it's safer to have this return `Instant?` and have the caller handle the case when there's no record yet.

Reply:
If you look in the `store` method, we always insert a row when we initially create the delivery config